contracts/multisig-prover/src/encoding/bcs.rs

```diff
@@ -0,0 +1,144 @@
+use bcs::to_bytes;
+use cosmwasm_std::{HexBinary, Uint256};
+use itertools::Itertools;
+use multisig::{key::Signature, msg::Signer};
+
+use crate::{error::ContractError, types::Operator};
+
+fn u256_to_u128(val: Uint256) -> u128 {
+    assert!(
+        val <= Uint256::from(u128::MAX),
+        "couldn't convert u256 ({}) to u128",
+        val
+    );
+    u128::from_le_bytes(
+        val.to_le_bytes()[..16]
+            .try_into()
+            .expect("couldn't convert u256 to u128"),
+    )
+}
+
+#[allow(dead_code)]
+fn encode_proof(
+    quorum: Uint256,
+    signers: Vec<(Signer, Option<Signature>)>,
+) -> Result<HexBinary, ContractError> {
+    let mut operators = make_operators_with_sigs(signers);
+    operators.sort();
+
+    let (addresses, weights, signatures): (Vec<_>, Vec<u128>, Vec<_>) = operators
```

Comment:
do you need `u128` here? Is the signature of `u256_to_u128` not enough for the compiler?